examples/with-cookie-auth/www/pages/login.js

```diff
@@ -0,0 +1,95 @@
+import { Component } from 'react'
+import Layout from '../components/layout'
+import { login } from '../utils/auth'
+
+class Login extends Component {
+  getInitialProps ({ req }) {
+    const apiUrl = `https://${req.headers.host}/api/profile.js`
+
+    return { apiUrl }
+  }
+
+  constructor (props) {
+    super(props)
+
+    this.state = { username: '', error: '' }
+    this.handleChange = this.handleChange.bind(this)
+    this.handleSubmit = this.handleSubmit.bind(this)
+  }
```

Comment:
You can get rid of the constructor and use property initializer and field arrow functions, like this: (Though i havent checked what is the minimal ES version)
```js
class Login {
  state = { username: '', error: '' }
  handleChange = (event) => {
    this.setState({ username: event.target.value });
    // this is always bound to this class
  }
```

Reply:
Is there a benefit of using one vs the other?